helm/prometheus-operator-app/values.yaml

```diff
@@ -1,2715 +1,85 @@
-    scheme: ""
-
-    ## tlsConfig: TLS configuration to use when scraping the endpoint. For example if using istio mTLS.
-    ## Of type: https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#tlsconfig
-    tlsConfig: {}
-
-    bearerTokenFile:
-
-    ## Metric relabel configs to apply to samples before ingestion.
-    ##
-    metricRelabelings: []
-    # - action: keep
-    #   regex: 'kube_(daemonset|deployment|pod|namespace|node|statefulset).+'
-    #   sourceLabels: [__name__]
-
-    #   relabel configs to apply to samples before ingestion.
-    ##
-    relabelings: []
-    # - sourceLabels: [__meta_kubernetes_pod_node_name]
-    #   separator: ;
-    #   regex: ^(.*)$
-    #   targetLabel: nodename
-    #   replacement: $1
-    #   action: replace
-
-  ## Settings affecting prometheusSpec
-  ## ref: https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#prometheusspec
-  ##
-  prometheusSpec:
-    ## If true, pass --storage.tsdb.max-block-duration=2h to prometheus. This is already done if using Thanos
-    ##
-    disableCompaction: false
-    ## APIServerConfig
-    ## ref: https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#apiserverconfig
-    ##
-    apiserverConfig: {}
-
-    ## Interval between consecutive scrapes.
-    ## Defaults to 30s.
-    ## ref: https://github.com/prometheus-operator/prometheus-operator/blob/release-0.44/pkg/prometheus/promcfg.go#L180-L183
-    ##
-    scrapeInterval: ""
-
-    ## Number of seconds to wait for target to respond before erroring
-    ##
-    scrapeTimeout: ""
-
-    ## Interval between consecutive evaluations.
-    ##
-    evaluationInterval: ""
-
-    ## ListenLocal makes the Prometheus server listen on loopback, so that it does not bind against the Pod IP.
-    ##
-    listenLocal: false
-
-    ## EnableAdminAPI enables Prometheus the administrative HTTP API which includes functionality such as deleting time series.
-    ## This is disabled by default.
-    ## ref: https://prometheus.io/docs/prometheus/latest/querying/api/#tsdb-admin-apis
-    ##
-    enableAdminAPI: false
-
-    ## WebTLSConfig defines the TLS parameters for HTTPS
-    ## ref: https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#webtlsconfig
-    web: {}
-
-    # EnableFeatures API enables access to Prometheus disabled features.
-    # ref: https://prometheus.io/docs/prometheus/latest/disabled_features/
-    enableFeatures: []
-    # - exemplar-storage
-
-    ## Image of Prometheus.
-    ##
+    prometheusSpec:
+      image:
+        repository: quay.io/giantswarm/prometheus
+    ingress:
+      ingressClassName: nginx
+      pathType: ImplementationSpecific
+    ingressPerReplica:
+      ingressClassName: nginx
+      pathType: ImplementationSpecific
+    thanosIngress:
+      ingressClassName: nginx
+      pathType: ImplementationSpecific
+  prometheusOperator:
```

Comment:
That's a default, no need to override here

Reply:
which section?

Reply:
enabled: true is a default value

Reply:
this is causes this problem:
```
[ERROR] templates/: template: prometheus-operator-app/templates/networkpolicy-admission.yaml:1:285: executing "prometheus-operator-app/templates/networkpolicy-admission.yaml" at <index .Values "prometheus-operator-app" "prometheusOperator" "admissionWebhooks" "certManager" "enabled">: error calling index: index of nil pointer
```

Reply:
because of this check
```
{{- if and  (index .Values "prometheus-operator-app" "prometheusOperator" "enabled") (index .Values "prometheus-operator-app" "prometheusOperator" "admissionWebhooks" "enabled") (index .Values "prometheus-operator-app" "prometheusOperator" "admissionWebhooks" "patch" "enabled") (not (index .Values "prometheus-operator-app" "prometheusOperator" "admissionWebhooks" "certManager" "enabled")) }}

```

Reply:
why does it uses index .Values "prometheus-operator-app" "prometheusOperator" "admissionWebhooks" "certManager" "enabled" and not index .Values "prometheusOperator" "admissionWebhooks" "certManager" "enabled" ?

Reply:
Keep it for now then

Reply:
this is for `NetworkPolicy` admission

Reply:
Yes it will go away in a future release then

Reply:
Fine for now